spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java

```diff
@@ -704,9 +763,49 @@ public void addFeatureSource(String layer, Map<String,String>  fields) {
                     cursor.close();
                 }
             }
+        } else {
+            try {
+                cursor = db.query(String.format("PRAGMA table_info(%s)", auditTableName));
+                ArrayList<String> existingCols = new ArrayList<>();
+                while (cursor.moveToNext()) {
+                    String columnName = SCSqliteHelper.getString(cursor, "name");
+                    existingCols.add(columnName);
+                }
+
+                Map<String,String> newSchemaCols = new HashMap<>();
+                newSchemaCols.putAll(fields);
+
+                for (String col: existingCols) {
+                    newSchemaCols.remove(col);
+                }
+
+                for (Map.Entry<String, String> col : newSchemaCols.entrySet())
+                {
+                    cursor = db.query(String.format("ALTER TABLE %s ADD COLUMN %s %s",
+                            auditTableName, col.getKey(), col.getValue())
+                    );
+                    cursor.moveToFirst();
+                }
+                tx.markSuccessful();
+            } catch (Exception e) {
+                Log.e(LOG_TAG, "Error adding to existing audit table: " + e.getMessage());
+            }
+            finally {
+                tx.end();
+                refreshFeatureSources();
+                if (cursor != null) {
+                    cursor.close();
+                }
+            }
+
         }
     }
 
+    public void addFeatureSource(String layer, Map<String,String>  fields) {
+        addContentsTable(layer, fields);
+        addAuditTable(layer, fields);
```

Comment:
`addAuditTable` should probably only run if `addContentsTable` is successful. Looks like Audit could run if the contents function fails.